src/analysis/inner_product_space/gram_schmidt_ortho.lean

```diff
@@ -0,0 +1,342 @@
+/-
+Copyright (c) 2022 Jiale Miao. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jiale Miao, Kevin Buzzard
+-/
+
+import analysis.inner_product_space.projection
+
+/-!
+# Gram-Schmidt Orthogonalization and Orthonormalization
+
+In this file we introduce Gram-Schmidt Orthogonalization and Orthonormalization.
+
+The Gram-Schmidt process takes a set of vectors as input
+and outputs a set of orthogonal vectors which have the same span.
+
+## Main results
+
+- `gram_schmidt_process` : Gram-Schmidt process
+- `gram_schmidt_process_orthogonal` :
+  the proof that "gram_schmidt_process" produces an orthogonal system of vectors
+- `span_gram_schmidt_process` :
+  Gram-Schmidt process preserves span of vectors
+- `gram_schmidt_process_ne_zero` :
+  If the input of first n vectors of gram_schmidt_process are linearly independent
+  , then output of first n vectors are non-zero
+- `gram_schmidt_process_normed` :
+  Normalized "Gram-Schmidt" (i.e each vector in this system has unit length)
+- `gram_schmidt_process_orthornormal` :
+  the proof that "gram_schmidt_process_normed" produces an orthornormal system of vectors
+
+## TODO
+  Construct a version with an orthonormal basis from the Gram-Schmidt process.
+-/
+
+open_locale big_operators
+
+variables (𝕜 : Type*) {E : Type*} [is_R_or_C 𝕜] [inner_product_space 𝕜 E]
+
+local notation `⟪`x`, `y`⟫` := @inner 𝕜 _ _ x y
+
+/-- Gram-Schmidt process -/
+noncomputable def gram_schmidt_process (f : ℕ → E) : ℕ → E
```

Comment:
For the sake of brevity, I suggest to drop `_process` from all the names.

Reply:
oh yes, that becomes more clear now, I have changed it, thank you!